crates/ruff_server/src/session/index/ruff_settings.rs

```diff
@@ -144,6 +161,8 @@ impl RuffSettingsIndex {
         let walker = builder.build_parallel();
 
         let index = std::sync::RwLock::new(index);
+        let error: std::sync::Mutex<bool> = std::sync::Mutex::new(error);
```

Comment:
You can use an `AtomicBool` instead of a `std::sync::Mutex`. An `AtomicBool` is more lightweight than a `Mutex`.